federation/pkg/federation-controller/sync/controller.go

```diff
@@ -176,9 +176,15 @@ func newFederationSyncController(client federationclientset.Interface, adapter f
 			return err
 		},
 		func(client kubeclientset.Interface, obj pkgruntime.Object) error {
+			secret := obj.(*apiv1.Secret)
 			namespacedName := adapter.NamespacedName(obj)
 			orphanDependents := false
 			err := adapter.ClusterDelete(client, namespacedName, &metav1.DeleteOptions{OrphanDependents: &orphanDependents})
+			// IsNotFound error is fine since that means the object is deleted already.
+			if errors.IsNotFound(err) {
+				glog.V(4).Infof("Secret %s/%s no longer exists", secret.Namespace, secret.Name)
```

Comment:
```golang
glog.V(4).Infof("%s %q no longer exists",  strings.ToTitle(adapter.Kind()), namespacedName)
```
The name of the type being federated can be retrieved via ``adapter.Kind()`` (returns lowercase), and namespacedName's ``String`` method results in ``[namespace]/[name]``.